core/src/main/java/co/cask/wrangler/steps/transformation/functions/GeoFences.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.steps.transformation.functions;
+
+import com.github.filosganga.geogson.gson.GeometryAdapterFactory;
+import com.github.filosganga.geogson.model.Coordinates;
+import com.github.filosganga.geogson.model.Feature;
+import com.github.filosganga.geogson.model.FeatureCollection;
+import com.github.filosganga.geogson.model.Polygon;
+import com.github.filosganga.geogson.model.positions.SinglePosition;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.util.List;
+
+/**
+ * GeoFencing check based on location and polygon
+ */
+public final class GeoFences {
+
+  private static final Gson GSON = new GsonBuilder()
+      .registerTypeAdapterFactory(new GeometryAdapterFactory())
+      .create();
+
+  /**
+   * Static method to be used with jexl
+   * Checks if Point is inside any of the given polygonal geofences based on the winding number algorithm.
+   *
+   * @param latitude  latitude of the location to verify
+   * @param longitude longitude of the location to verify
+   * @param geofences GeoJson representation of the fence area
+   * @return true if location is inside any of the given geofences, else false
+   */
+  public static Boolean inFence(double latitude, double longitude, String geofences) {
+    Coordinates location = Coordinates.of(longitude, latitude);
+    Boolean inzone = false;
+    FeatureCollection featureCollection;
+    try {
+      featureCollection = GSON.fromJson(geofences, FeatureCollection.class);
+    } catch (IllegalArgumentException e) {
```

Comment:
The Preconditions.checkArgument() method throws the `IllegalArgumentException` How ever i wrapped it to make it more user friendly because the method has no error message and it can be confusing.

The checkArgument() is called by the geoGson library while parsing the json as polygon type object
https://github.com/filosganga/geogson/blob/master/core/src/main/java/com/github/filosganga/geogson/model/Polygon.java#L49